hollow-test/src/main/java/com/netflix/hollow/test/HollowReadStateEngineBuilder.java

```diff
@@ -0,0 +1,67 @@
+package com.netflix.hollow.test;
+
+import com.netflix.hollow.core.read.engine.HollowReadStateEngine;
+import com.netflix.hollow.core.util.StateEngineRoundTripper;
+import com.netflix.hollow.core.write.HollowWriteStateEngine;
+import com.netflix.hollow.core.write.objectmapper.HollowObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * This class allows constructing HollowReadStateEngine instances for testing purposes.
+ * You cannot add to a HollowReadStateEngineBuilder after calling build, however you may build
+ * multiple HollowReadStateEngine objects with the same builder.
+ */
+public class HollowReadStateEngineBuilder {
+    private final HollowWriteStateEngine writeEngine;
+    private final HollowObjectMapper objectMapper;
+
+    private boolean built;
+
+    /**
+     * Create a HollowReadStateEngineBuilder with an empty initial type state. Adding objects will
+     * add their types, if not already present.
+     */
+    public HollowReadStateEngineBuilder() {
+        this(new ArrayList<Class<?>>());
```

Comment:
Minor: Collectons.emptyList() ?

Reply:
Sure, will do. Note to self: make sure to specify the type ie `Collections.<Class<?>>.emptyList()`